### ISSUE TITLE
"nuxt generate" no longer produces index.html

### ISSUE THREAD

**Issue by teajaymars** [nuxt/nuxt#28042]
### Environment

Working directory: /Users/zephod/code/dp/tmp6
Nuxt project info: (copied to clipboard)

------------------------------
- Operating System: Darwin
- Node Version:     v18.17.1
- Nuxt Version:     3.12.0
- CLI Version:      3.12.0
- Nitro Version:    2.9.7
- Package Manager:  yarn@1.22.4
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

```bash
nuxi create tmp
cd tmp
yarn
yarn generate
ls .output/public
```

Result:

```
200.html    404.html    _nuxt       favicon.ico
```

### Describe the bug

Since Nuxt 3.12.0, the `nuxt generate` command does not work as documented.

* This occurs even using the simplest template from `nuxi create`
* It says "✔ You can now deploy .output/public to any static hosting!"
* The .output/public folder does not contain an index.html file.

I've experimented with alternative structures: Adding pages, adding layouts, removing app.vue. Still no static .html files are generated, except for 200.html and 404.html.

Reverting to Nuxt 3.11.2, deleting yarn.lock and rerunning `yarn && yarn generate` fixes the problem.

**Comment by manniL** [comment]
Duplicate of https://github.com/nuxt/nuxt/issues/27694

**Comment by teajaymars** [comment]
(Thanks for reviewing so fast!)
However -- not a dupe, this is a live bug in 3.12.3. I think the fix in #27694 is a partial fix. Just doing some more testing on 3.12.3:

* Since 3.12.3, it **works** if you create `pages/index.vue`.
* However, the default case (`app.vue` alone) still does not work.

Shortest bug description I can give you is that right now, with 3.12.3, this workflow is broken:

```bash
nuxi create tmp
cd tmp && yarn && yarn generate
```

It will say "You can now deploy .output/public to any static hosting" but you cannot. I guess that the simplest layout (just `app.vue`, no `pages/` folder) is not supported by the static site generator.

**Comment by manniL** [comment]
That's true! 
Let's track in https://github.com/nuxt/nuxt/issues/28037 for now (as it is open already)
